lib/BackbeatConsumer.js

```diff
@@ -783,9 +784,31 @@ class BackbeatConsumer extends EventEmitter {
         }
         this._circuitBreaker.stop();
         return async.series([
+            next => {
+                // stopping consumption of new messages while
+                // waiting for current ones to be processed
+                this._isShuttingDown = true;
+                // waiting for all current jobs to end before exiting
+                if (this._processingQueue && !this._processingQueue.idle()) {
+                    this._processingQueue.drain = () => {
```

Comment:
* to be on the safe side, we should reset the drain callback once it is called
* what happens if there is a rebalance during shutdown? we may have an issue, since we overwrite the callback there as well... :-/

Reply:
Since we block the polls when shutting down, the rebalance callback will not be called.

If a rebalance was already in progress, the drain function will be overwritten, and as soon as the jobs finish the consumer will be disconnected, maybe we'll have issues committing the offsets tho..

Reply:
use the rebalance callback to handle all the logic, we shouldn't have these problems anymore